src/EditorFeatures/Core/Shared/Extensions/DocumentExtensions.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.Shared.Extensions
+{
+    internal static class DocumentExtensions
+    {
+        public static bool IsInCloudEnvironmentClientContext(this Document document)
```

Comment:
could this go in one of the existing ones?
maybe http://sourceroslyn.io/#Microsoft.CodeAnalysis.Workspaces/DocumentExtensions.cs,c73098348ffcde85

Reply:
I tried the existing ones - it looks like either IWorkspaceContextService isn't available in the layer (like the one you linked above), or the particular DocumentExtensions isn't accessible from the callsites :/